corehq/ex-submodules/phonelog/models.py

```diff
@@ -19,6 +19,9 @@ class DeviceReportEntry(models.Model):
 
     class Meta:
         unique_together = [('xform_id', 'i')]
+        index_together = [
+            ("date", "domain"),
```

Comment:
I saw your note about your reasoning for making `date` the major index, but I don't think it's right. The query planner decided that if it needs to choose between filtering by date range using an index and doing a sequential scan to filter by domain, and filtering by domain using an index and then doing a sequential scan to filter by date range—it chose to do it by date first, probably because it expected that to pare it down more than the alternative. (I.e. it expected that "all logs in domain foo" would return more results than "all logs in the last month"—which may or may not be true depending on the domain.) But the whole point of making an index is that with this index, the planner doesn't need to make this choice anymore—it can satisfy the entire query with a single index range lookup in a btree!

Disclaimer: I don't actually know anything about planners or how they work—this is based more on from scratch reasoning than on actual knowledge